tree/dataframe/src/RSqliteDS.cxx

```diff
@@ -13,19 +13,304 @@
     \ingroup dataframe
     \brief RDataFrame data source class for reading SQlite files.
 */
+
 // clang-format on
 
-#include <ROOT/RSqliteDS.hxx>
-#include <ROOT/RDFUtils.hxx>
-#include <ROOT/RMakeUnique.hxx>
+#include "ROOT/RSqliteDS.hxx"
+
+#include "ROOT/RConfig.h"
+#include "ROOT/RDFUtils.hxx"
+#include "ROOT/RMakeUnique.hxx"
 
-#include <TError.h>
+#include "TError.h"
+#include "TRandom.h"
+#include "TSystem.h"
 
 #include <algorithm>
 #include <cctype>
 #include <cstring>
+#include <ctime>
+#include <memory> // for placement new
 #include <stdexcept>
 
+#ifdef R__HAS_DAVIX
+#include <davix.hpp>
+#endif
+
+namespace {
+
+constexpr char const *gSQliteVfsName = "ROOT-Davix-readonly";
+
+#ifdef R__HAS_DAVIX
+
+/**
+ * The "derived class" from sqlite3_file that gets passed to the file operation callbacks.
+ */
+struct VfsRootFile {
+   VfsRootFile() : pos(&c) {}
+   sqlite3_file pFile;
+   DAVIX_FD *fd;
+   uint64_t size;
+   Davix::Context c;
+   Davix::DavPosix pos;
+};
+
+////////////////////////////////////////////////////////////////////////////
+/// Releases the resources associated to a file opened with davix
+static int VfsRdOnlyClose(sqlite3_file *pFile)
+{
+   Davix::DavixError *err = nullptr;
+   VfsRootFile *p = reinterpret_cast<VfsRootFile *>(pFile);
+   if (p->pos.close(p->fd, &err) == -1) {
+      p->~VfsRootFile();
+      return SQLITE_IOERR_CLOSE;
+   }
+   p->~VfsRootFile();
+   return SQLITE_OK;
+}
+
+////////////////////////////////////////////////////////////////////////////
+/// Issues an HTTP range request to read a chunk from a remote file
+static int VfsRdOnlyRead(sqlite3_file *pFile, void *zBuf, int count, sqlite_int64 offset)
+{
+   Davix::DavixError *err = nullptr;
+   VfsRootFile *p = reinterpret_cast<VfsRootFile *>(pFile);
+   if (p->pos.pread(p->fd, zBuf, count, offset, &err) == -1) {
+      return SQLITE_IOERR;
+   }
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyWrite(sqlite3_file * /*pFile*/, const void * /*zBuf*/, int /*iAmt*/, sqlite_int64 /*iOfst*/)
+{
+   return SQLITE_OPEN_READONLY;
+}
```

Comment:
as a non-sqlite expert, I really need a comment in this function and the following to understand what's going on :)